docs/basics/101-133-containersrun.rst

```diff
@@ -0,0 +1,306 @@
+on your own machine once a new software update is installed. Therefore, it is
+important to communicate all details about the computational environment for
+an analysis as thoroughly as possible. Luckily, DataLad provides an extension
+that can link computational environments to datasets, the
+`datalad containers <http://docs.datalad.org/projects/container/en/latest/>`_
+extension [#f1]_.
+
+This section will give a quick overview on what containers are and
+demonstrate how ``datalad-containers`` helps to capture full provenance of an
+analysis by linking containers to datasets and analyses.
+
+Containers
+^^^^^^^^^^
+
+To put it simple, computational containers are cut-down virtual machines that
+allow you to package all software libraries and their dependencies (all in the
+precise version your analysis requires) into a bundle you can share with
+others. On your own and others machines, the container constitutes a secluded
+software environment that
+
+- contains the exact software environment that you specified, ready to run
+  analyses in
+- does not effect any software outside of the container
+
+Unlike virtual machines, software containers do not have their own operating
+system. Instead, they use basic services of the underlying operating system
+of the computer they run on (in a read-only fashion). This makes them
+lightweight and portable. By sharing software environments with containers,
+others (and also yourself) have easy access to the correct software
+without the need to modify the software environment of the machine the
+container runs on. Thus, containers are ideal to encapsulate the software
+environment and share it together with the analysis code and data to ensure
+computational reproducibility of your analyses.
+
+There are a number of different tools to create and use containers, with
+`Docker <https://www.docker.com/>`_ being one of the most well-known of them.
+While being a powerful tool, it can not be used on high performance computing
+(HPC) infrastructure [#f2]_. An alternative is `Singularity <https://sylabs
+.io/docs/>`_.
+Both of these tools share core terminology:
+
+**Image**
+   A template or "recipe" from which containers are build. It lists all
+   required components of the computational environment, and is made by a
+   human user. If you want to create your own container, you start by writing
+   an Image file, but you can also *pull* a publicly shared image from the
+   *Hub* of the tool you are using. Based on images, containers are *build*.
+
+**Container**
+  The instance that you can actually use for your computations, *build* from
+  an image.
+
+**Hub**
+  A storage resource to share and consume images. There is
+  `Singularity-Hub <https://singularity-hub.org/>`_ and
+  `Docker-Hub <https://hub.docker.com/>`_.
+
+Note that as of now, the ``datalad-containers`` extensions only supports
+Singularity images, but support for Docker is being actively developed.
```

Comment:
ahh, thats news to me. I misread this paragraph in the datalad-containers docs:

> At the moment there is built-in support for Singularity images, but other container execution systems can be used together with custom helper scripts. Direct support for Docker is under development.

In this case, I should learn how to use Docker with ``datalad containers-run``, and include it as well (maybe hidden)

Reply:
@mih
> I think Docker works fine. @kyleam do you think otherwise?

There are known rough edges and probably a good deal of unknown ones given that the docker adapter hasn't seen heavy use, but the basic idea is there and works at least for simple cases.

@adswa
>  I misread this paragraph in the datalad-containers docs:

No, I don't think you misread it.  That's just stale.  It was added in f9cecf2 (DOC: Basic intro/demo (fixes gh-4), 2018-05-28).  The docker adapter wasn't in the tree then:

```
$ git ls-tree -r f9cecf2 | grep -c docker
0
```

I'll update that text.

Reply:
Thx, @kyleam!